tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,635 @@
+        #sort array
+        outcome.sort(axis=0, order=list(dicts))
+        #create dataframe
+        final = pd.DataFrame.from_records(outcome)
+    return final
+
+
+#this is for dataframes/series
+#this should be the final output if there are lists or lists and dicts
+#returned from check instance
+def compute_two_dfs(df1, df2):
+    #get lengths of dataframes(number of rows) and swap
+    # essentially we'll pair one dataframe with the other's length:
+    lengths = reversed([ent.index.size for ent in (df1, df2)])
+    #grab the maximum string length
+    string_cols = [
+        frame.select_dtypes(include="object").columns for frame in (df1, df2)
+    ]
+
+    #pair max string length with col
+    #will be passed into frame.to_records, to get dtype in numpy recarray
+    string_cols = [{col: f"<U{frame[col].str.len().max()}"
+                    for col in ent}
+                   for ent, frame in zip(string_cols, (df1, df2))]
+
+    (len_first, col_dtypes,
+     first), (len_last, col_dtypes,
+              last) = list(zip(lengths, string_cols, (df1, df2)))
+
+    #export to numpy as recarray
+    first = first.to_records(column_dtypes=col_dtypes, index=False)
+    #tile first with len_first
+    #remember, len_first is the length of the other dataframe
+    first = np.tile(first, (len_first, 1))
+    #get a 1d array
+    first = np.concatenate(first)
+    #sorting here ensures we get each row of the first
+    #with the entire rows of the other dataframe
+    np.recarray.sort(first, order=first.dtype.names[0])
+
+    #same process as first, except there'll be no sorting
+    last = last.to_records(column_dtypes=col_dtypes, index=False)
+    last = np.tile(last, (len_last, 1))
+    last = np.concatenate(last)
+    result = rfn.merge_arrays((first, last), flatten=True, asrecarray=True)
+    return pd.DataFrame.from_records(result)
+
+
+def grid_computation_list(dfs):
+    return reduce(compute_two_dfs, dfs)
+
+
+def grid_computation(dfs, dicts):
+    if not dicts:
+        result = grid_computation_list(dfs)
+    elif not dfs:
+        result = grid_computation_dict(dicts)
+    else:
+        dfs.append(grid_computation_dict(dicts))
+        result = grid_computation_list(dfs)
+    return result
+
+
+########################### Tests
+
+
+def test_not_a_dict():
+    """Test that entry(list) is not a dictionary"""
+    data = [60, 70]
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_not_a_dict_1():
+    """Test that entry (dataframe) is not a dictionary"""
+    data = pd.DataFrame([60, 70])
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_empty_dict():
+    """Test that entry should not be empty"""
+    data = {}
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_scalar_to_list():
+    """Test that 
+    scalars(null,boolean,int,float), strings, sets, and tuples 
+    are converted to lists in the dictionary values
+     """
```

Comment:
I believe this might be a more concise description of what's happening.

```suggestion
    """
    Test that dictionary values are all converted to lists.
    """
```